lib/LTP/WhiteList.pm

```diff
@@ -126,10 +126,16 @@ sub override_known_failures {
         }
     }
 
-    my $msg = "Failure in LTP:$suite:$testname is known, overriding to softfail";
-    bmwqemu::diag($msg);
-    $testmod->{result} = 'softfail';
-    $testmod->record_soft_failure_result(join("\n", $msg, ($entry->{message} // ())));
+    if ($entry->{keep_fail}) {
+        $testmod->record_resultfile('Known', $entry->{message}, result => 'fail');
```

Comment:
@Martchus @mdoucha I was trying to find, why `Cannot remove memory cgroup in subtest 3. bsc#1214890` in `keep_fail` message in https://openqa.suse.de/tests/12538408#step/memcg_regression/12 does not perform URL transformation [`bugref_to_href`](https://github.com/os-autoinst/openQA/blob/master/lib/OpenQA/Utils.pm#L456-L461), which soft failures do (https://openqa.suse.de/tests/12497280#step/ustat01/12).

Soft failures use [`record_soft_failure_result`](https://github.com/os-autoinst/os-autoinst/blob/master/basetest.pm#L466-L475), but that is quite similar to [`record_resultfile`](https://github.com/os-autoinst/os-autoinst/blob/master/basetest.pm#L436-L452). 
What am I missing?

Reply:
That's strange, indeed. I'll have a look.

Reply:
So the problem is this JavaScript code which doesn't handle rendering bugrefs: https://github.com/os-autoinst/openQA/blob/f46075b5f42985c47976249f44935c80fa780bca/assets/javascripts/render.js#L187

The problem with just implementing it is that we'd have to duplicate/rewrite existing server-side Perl code in JavaScript which wouldn't be ideal. I guess ideally all that frontend logic would be done entirely in JavaScript.

Reply:
Thanks for finding the root cause. Hm, long time ago web developers tried to avoid text handling in javascript. But openQA web UI is simply unusable without javascript (job detail shows nothing), thus we can ignore bug reference being broken without javascript. But somebody would have to implement it.

Reply:
It looks like this function is actually quite simple - only the regex it uses is a bit involved. So duplicating the function in JavaScript wouldn't be that hard and the regex could simply be made available to the JavaScript like other backend variables. I'll give it a try :-)

Reply:
I tried the following: https://github.com/Martchus/openQA/pull/new/render-bugrefs-in-js-as-well

But it doesn't work because the JavaScript regex support cannot cope with the Perl regex we're using in the backend.

Reply:
Thank you both. I'm not familiar with the code, but I'd also prefer Martin's suggestion.

Reply:
Mh… it is not that horrible but I was also thinking about simply rendering the HTML server-side.

Reply:
This would improve the rendering: https://github.com/os-autoinst/openQA/pull/5334

Getting rid of the hack would also be a possibility but it would mean a slight waiting time when loading a step which can be annoying when going though a lot of them so I refrained from that for now.